LoopKitUI/Views/CustomAlert.swift

```diff
@@ -0,0 +1,130 @@
+//
+//  CustomAlert.swift
+//  LoopKit
+//
+//  Created by Cameron Ingham on 7/17/24.
+//  Copyright © 2024 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+public struct CustomAlert {
+    public enum ImageOption {
+        case warning
+        case custom(
+            UIImage?,
+            bounds: (NSTextAttachment) -> CGRect
+        )
+    }
+    
+    let image: ImageOption?
+    let title: String?
+    let message: String?
+    let primaryAction: UIAlertAction?
+    let secondaryAction: UIAlertAction?
+    
+    public init(
+        image: ImageOption?,
+        title: String?,
+        message: String?,
+        primaryAction: UIAlertAction?,
+        secondaryAction: UIAlertAction?
+    ) {
+        self.image = image
+        self.title = title
+        self.message = message
+        self.primaryAction = primaryAction
+        self.secondaryAction = secondaryAction
+    }
+}
+
+public struct CustomAlertPresenter: UIViewControllerRepresentable {
+    
+    @Environment(\.guidanceColors) private var guidanceColors
+    
+    public class CustomAlertPresentingViewController: UIViewController {
+        private var alertController: UIAlertController?
+        
+        private let alertBinding: Binding<CustomAlert?>
+        private let guidanceColors: GuidanceColors
+        
+        fileprivate init(alertBinding: Binding<CustomAlert?>, guidanceColors: GuidanceColors) {
+            self.alertBinding = alertBinding
+            self.guidanceColors = guidanceColors
```

Comment:
not blocking but curious why the `guidanceColors` is both pulled from the environment and passed in for initialization?

Reply:
I don't think `guidanceColors` gets picked up by a `UIViewController` just the wrapping `UIViewControllerRepresentable` which is a SwiftUI view. I can test this though.